### PR TITLE
feat(wallet-dashboard): add tx details state to the end of send visual asset flow

### LINKED ISSUE #4256: Add tx details state to the end of send visual asset flow
Add tx details state to the end of send visual asset flow, take wallet as reference
Depends on https://github.com/iotaledger/iota/issues/4245

### PR DESCRIPTION
**Pull request by panteleymonchuk**
## Links to any relevant issues

Closes #4256
Depends on: https://github.com/iotaledger/iota/pull/4422

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

Send a visual asset to some address.

Expected behaviour: user is able to see recipt.
image

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: apps/wallet-dashboard/components/Dialogs/Assets/AssetDialog.tsx [hunk 1]
```diff
@@ -44,10 +48,15 @@ export function AssetDialog({ onClose, asset }: AssetsDialogProps): JSX.Element
...
+
+            const tx = await iotaClient.waitForTransaction({
+                digest: executed.digest,
+            });
+
+            setDigest(tx.digest);
             addNotification('Transfer transaction successful', NotificationType.Success);
-            onClose();
-            setView(AssetsDialogView.Details);
+            setView(AssetsDialogView.Recipt);
```
Comment (begonaalvarezd) [inline apps/wallet-dashboard/components/Dialogs/Assets/AssetDialog.tsx]: Spellcheck is failing, it should be `Receipt`

**Review by begonaalvarezd — CHANGES_REQUESTED** [review]

File: apps/wallet-dashboard/components/Dialogs/Assets/AssetDialog.tsx (hunk 1 above)
Comment (begonaalvarezd) [reply apps/wallet-dashboard/components/Dialogs/Assets/AssetDialog.tsx]: In order PRs I see we are calling this `TransactionDetails` though, we should unify it

**Review by begonaalvarezd — CHANGES_REQUESTED** [review]

**Review by evavirseda — APPROVED** [review]

**Review by cpl121 — CHANGES_REQUESTED** [review]
build error 🙏🏼

File: apps/wallet-dashboard/components/Dialogs/Assets/AssetDialog.tsx [hunk 2]
```diff
@@ -44,10 +48,15 @@ export function AssetDialog({ onClose, asset }: AssetsDialogProps): JSX.Element
...
     async function onSubmit(values: FormValues) {
         try {
-            await sendAsset.mutateAsync(values.to);
+            const executed = await sendAsset.mutateAsync(values.to);
+
+            const tx = await iotaClient.waitForTransaction({
+                digest: executed.digest,
+            });
+
+            setDigest(tx.digest);
```
Comment (begonaalvarezd) [inline apps/wallet-dashboard/components/Dialogs/Assets/AssetDialog.tsx:58]: I think we should refetch at this point

**Review by begonaalvarezd — CHANGES_REQUESTED** [review]

File: apps/wallet-dashboard/components/Dialogs/Assets/AssetDialog.tsx [hunk 3]
```diff
@@ -66,16 +77,30 @@ export function AssetDialog({ onClose, asset }: AssetsDialogProps): JSX.Element
...
+                                onBack={onSendViewBack}
+                            />
+                        )}
+
+                        {view === AssetsDialogView.TransactionDetails && !!digest ? (
+                            <TransactionDetailsView digest={digest} onClose={onOpenChange} />
+                        ) : null}
+                    </>
+                </FormikProvider>
+            </DialogLayout>
```
Comment (cpl121) [inline apps/wallet-dashboard/components/Dialogs/Assets/AssetDialog.tsx]: we don't need views `DetailsView` and `TransactionDetailsView` inside the Formik Provider
```suggestion
           <DialogLayout>
                {view === AssetsDialogView.Details && (
                    <DetailsView asset={asset} onClose={onOpenChange} onSend={onDetailsSend} />
                )}
                {view === AssetsDialogView.Send && (
                    <FormikProvider value={formik}>
                        <SendView asset={asset} onClose={onOpenChange} onBack={onSendViewBack} />
                    </FormikProvider>
                )}
                {view === AssetsDialogView.TransactionDetails && !!digest ? (
                    <TransactionDetailsView digest={digest} onClose={onOpenChange} />
                ) : null}
            </DialogLayout>
```

**Review by cpl121 — CHANGES_REQUESTED** [review]

**Review by begonaalvarezd — APPROVED** [review]

**Review by cpl121 — APPROVED** [review]

**Pull request merged**